polygeist/test/polygeist-opt/sycl/kernel_disjoint_specialization.mlir

```diff
@@ -0,0 +1,110 @@
+  // CHECK-NEXT:    %c0_i32_4 = arith.constant 0 : i32
+  // CHECK-NEXT:    %4 = sycl.range.get %alloca_0[%c0_i32_4] {ArgumentTypes = [memref<1x!sycl_range_1_>, i32], FunctionName = @get, TypeName = @range} : (memref<1x!sycl_range_1_>, i32) -> index
+  // CHECK-NEXT:    memref.store %4, %3[%c0_1] : memref<?xindex>
+  // CHECK-NEXT:    [[ACC1_END:%.*]] = sycl.accessor.subscript %arg0[%alloca_2] {ArgumentTypes = [memref<?x!sycl_accessor_1_f32_r_gb>, memref<1x!sycl_id_1_>], FunctionName = @"operator[]", TypeName = @accessor} : (memref<?x!sycl_accessor_1_f32_r_gb>, memref<1x!sycl_id_1_>) -> memref<?xf32, 1>
+
+  // COM: Version with condition: [[ACC1_END]] <= [[ACC2_BEGIN]] || [[ACC1_BEGIN]] >= [[ACC2_END]].
+  // CHECK:         [[ACC2_BEGIN:%.*]] = sycl.accessor.subscript %arg1[{{.*}}] {ArgumentTypes = [memref<?x!sycl_accessor_1_f32_w_gb>, memref<1x!sycl_id_1_>], FunctionName = @"operator[]", TypeName = @accessor} : (memref<?x!sycl_accessor_1_f32_w_gb>, memref<1x!sycl_id_1_>) -> memref<?xf32, 1>
+  // CHECK:         [[ACC2_END:%.*]] = sycl.accessor.subscript %arg1[{{.*}}] {ArgumentTypes = [memref<?x!sycl_accessor_1_f32_w_gb>, memref<1x!sycl_id_1_>], FunctionName = @"operator[]", TypeName = @accessor} : (memref<?x!sycl_accessor_1_f32_w_gb>, memref<1x!sycl_id_1_>) -> memref<?xf32, 1>
+  // CHECK-DAG:     [[ACC1_END_PTR:%.*]] = "polygeist.memref2pointer"([[ACC1_END]]) : (memref<?xf32, 1>) -> !llvm.ptr<f32, 1>
+  // CHECK-DAG:     [[ACC2_BEGIN_PTR:%.*]]  = "polygeist.memref2pointer"([[ACC2_BEGIN]]) : (memref<?xf32, 1>) -> !llvm.ptr<f32, 1>
+  // CHECK-NEXT:    %14 = llvm.icmp "ule" [[ACC1_END_PTR]], [[ACC2_BEGIN_PTR]] : !llvm.ptr<f32, 1>
+  // CHECK-DAG:     [[ACC1_BEGIN_PTR:%.*]] = "polygeist.memref2pointer"([[ACC1_BEGIN]]) : (memref<?xf32, 1>) -> !llvm.ptr<f32, 1>
+  // CHECK-DAG:     [[ACC2_END_PTR:%.*]] = "polygeist.memref2pointer"([[ACC2_END]]) : (memref<?xf32, 1>) -> !llvm.ptr<f32, 1>
+  // CHECK-NEXT:    %17 = llvm.icmp "uge" [[ACC1_BEGIN_PTR]], [[ACC2_END_PTR]] : !llvm.ptr<f32, 1>
+  // CHECK-NEXT:    %18 = arith.ori %14, %17 : i1
+  // CHECK-NEXT:    scf.if %18 {
+  // CHECK-NEXT:      func.call @callee1.specialized(%arg0, %arg1) : (memref<?x!sycl_accessor_1_f32_r_gb>, memref<?x!sycl_accessor_1_f32_w_gb>) -> ()
+  // CHECK-NEXT:    } else {
+  // CHECK-NEXT:      func.call @callee1(%arg0, %arg1) : (memref<?x!sycl_accessor_1_f32_r_gb>, memref<?x!sycl_accessor_1_f32_w_gb>) -> ()
+  // CHECK-NEXT:    }
+  func.func private @callee1(%arg0: memref<?x!sycl_accessor_1_f32_r_gb>, %arg1: memref<?x!sycl_accessor_1_f32_w_gb>) {
+    return
+  }
+  gpu.func @caller1(%arg0: memref<?x!sycl_accessor_1_f32_r_gb>, %arg1: memref<?x!sycl_accessor_1_f32_w_gb>) kernel {
+    func.call @callee1(%arg0, %arg1) : (memref<?x!sycl_accessor_1_f32_r_gb>, memref<?x!sycl_accessor_1_f32_w_gb>) -> ()
+    gpu.return
+  }
+
+  // COM: No need to version as the %arg0 and %arg1 do not alias (from different malloc).
```

Comment:
I think this will lead to issues. We should check anyway. We could have:
```
%alloca = memref.alloca() : memref<1x!sycl_accessor_1_f32_r_gb>
%cast = memref.cast %alloca : memref<1x!sycl_accessor_1_f32_r_gb> to memref<?x!sycl_accessor_1_f32_r_gb>
%alloca_0 = memref.alloca() : memref<1x!sycl_accessor_1_f32_w_gb>
%cast_1 = memref.cast %alloca_0 : memref<1x!sycl_accessor_1_f32_w_gb> to memref<?x!sycl_accessor_1_f32_w_gb>
sycl.constructor @accessor(%cast, %buffer)
sycl.constructor @accessor(%cast_1, %buffer)
func.call @callee2(%cast, %cast_1) : (memref<?x!sycl_accessor_1_f32_r_gb>, memref<?x!sycl_accessor_1_f32_w_gb>) -> ()
```
As we can see, the constructor assigns the same buffer to both accessors, so that will lead to issues.

Reply:
Right, remove the change to use alias analysis, and remove this portion of the lit test.